vertx-oracle-client/src/main/java/io/vertx/oracleclient/impl/Helper.java

```diff
@@ -332,4 +344,34 @@ default void run() throws VertxException {
      */
     JdbcRow copy();
   }
+
+  @FunctionalInterface
+  public interface SQLBlockingCodeHandler<T> extends Handler<Promise<T>> {
```

Comment:
as this is blocker I think it would be better to have a synchronous return from doHandler, i.e 

```
T doHandle();
```

and let the `handle` implementation complete the promise.

Reply:
Good point. I've updated the PR.